packages/openapi-v3/src/enhancers/extensions/security.spec.extension.ts

```diff
@@ -6,37 +6,36 @@
 import {bind} from '@loopback/core';
 import debugModule from 'debug';
 import {inspect} from 'util';
-import {
-  mergeOpenAPISpec,
-  ReferenceObject,
-  SecuritySchemeObject,
-} from '../../../..';
-import {asSpecEnhancer, OASEnhancer} from '../../../../enhancers/types';
-import {OpenApiSpec} from '../../../../types';
+import {mergeOpenAPISpec, ReferenceObject, SecuritySchemeObject} from '../..';
+import {OpenApiSpec} from '../../types';
+import {asSpecEnhancer, OASEnhancer} from '../types';
 const debug = debugModule('loopback:openapi:spec-enhancer');
 
 export type SecuritySchemeObjects = {
   [securityScheme: string]: SecuritySchemeObject | ReferenceObject;
 };
 
 export const SECURITY_SCHEME_SPEC: SecuritySchemeObjects = {
-  bearerAuth: {
+  jwt: {
     type: 'http',
     scheme: 'bearer',
     bearerFormat: 'JWT',
   },
 };
 
 /**
- * A spec enhancer to add bearer token OpenAPI security entry to
+ * A spec enhancer to add jwt bearer token OpenAPI security entry to
  * `spec.component.securitySchemes`
  */
 @bind(asSpecEnhancer)
 export class SecuritySpecEnhancer implements OASEnhancer {
   name = 'security';
 
   modifySpec(spec: OpenApiSpec): OpenApiSpec {
-    const patchSpec = {components: {securitySchemes: SECURITY_SCHEME_SPEC}};
+    if (spec?.components?.securitySchemes?.jwt) return spec;
```

Comment:
I would expect the security spec enhancers to be contributed by authentication strategies instead of being hard-coded in `@loopback/rest` module. This code was added as part of the test fixture intentionally.

Reply:
Subject to the story(#4386) it is a default and I've opted to not override an existing item. @jannyHou What to do?

Reply:
@raymondfeng I think @dougal83 is adding the default authorizor button in explorer.

<img width="1035" alt="Screen Shot 2020-02-07 at 2 04 03 PM" src="https://user-images.githubusercontent.com/12554153/74057825-ba7d4680-49b2-11ea-90bb-c2304ee86895.png">
<img width="1249" alt="Screen Shot 2020-02-07 at 2 03 57 PM" src="https://user-images.githubusercontent.com/12554153/74057828-bbae7380-49b2-11ea-9ed3-c650f231a523.png">

Reply:
@raymondfeng I think we agree to not bind a built-in jwt strategy into LB4 app, so the only thing we can provide the the security spec, as what @dougal83 adds here.

How about introduce an option `useDefaultSecuritySpec`, sets to true by default so that our explorer shows the button. If people provide their own auth strategy & security spec enhancer , they can turn off the option.

Reply:
> I would expect the security spec enhancers to be contributed by authentication strategies instead of being hard-coded in `@loopback/rest` module.

+1

> How about introduce an option useDefaultSecuritySpec, sets to true by default so that our explorer shows the button. If people provide their own auth strategy & security spec enhancer , they can turn off the option.

As I mentioned in my other comment, it's my understanding that the REST component does not provide any authentication mechanism out of the box, there is no "default security". Therefore the option `useDefaultSecuritySpec` does not make sense to me.

IMO, the security spec should be contributed by `@loopback/authentication`, because that's the extension adding support for bearer tokens. The enhancer should be clever enough to handle the case when there are multiple authentication extensions and/or security specs involved (e.g. Bearer token, HTTP Basic/Digest auth, etc.) and preserve those additional security specs.

Reply:
> @raymondfeng I think @dougal83 is adding the default authorizor button in explorer.
> 
> <img alt="Screen Shot 2020-02-07 at 2 04 03 PM" width="1035" src="https://user-images.githubusercontent.com/12554153/74057825-ba7d4680-49b2-11ea-90bb-c2304ee86895.png">
> 
> <img alt="Screen Shot 2020-02-07 at 2 03 57 PM" width="1249" src="https://user-images.githubusercontent.com/12554153/74057828-bbae7380-49b2-11ea-9ed3-c650f231a523.png">

@jannyHou #4386 Having read the issue/story properly just now... I had made a big mistake on my understanding... I thought that is was generally adding a default to the rest server! 🤦‍♂ To make amends I guess I'll have to look into explorer and making the change as per the story!